File: src/main/java/cn/edu/tsinghua/sdfs/protocol/serilizer/Serializer.java
package cn.edu.tsinghua.sdfs.protocol.serilizer;


import cn.edu.tsinghua.sdfs.protocol.serilizer.impl.JSONSerializer;

public interface Serializer {

	Serializer DEFAULT = new JSONSerializer();

	byte[] serialize(Object object);

	<T> T deserialize(byte[] bytes, Class<? extends T> clazz);

}

File: src/main/java/cn/edu/tsinghua/sdfs/protocol/serilizer/Serializer.kt
package cn.edu.tsinghua.sdfs.protocol.serilizer


import cn.edu.tsinghua.sdfs.protocol.serilizer.impl.JSONSerializer

interface Serializer {

    fun serialize(obj: Any): ByteArray

    fun <T> deserialize(bytes: ByteArray, clazz: Class<out T>): T

    companion object {

        val DEFAULT: Serializer = JSONSerializer()
    }

}
